akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -6,10 +6,28 @@ package akka.stream
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
+/**
+ * An input port of a StreamLayout.Module. This type logically belongs
+ * into the impl package but must live here due to how `sealed` works.
```

Comment:
then it does not logically belong to the impl package, and I think this is more confusing than helpful for the user. Perhaps the reason should be in a non-scaladoc comment instead?